### PR TITLE
nordic: add support for nRF54 Series PWM instances

### PR DESCRIPTION
**Pull request by nika-nordic**
Draft until MDK fix in `hal_nordic` is available

### CODE REVIEW

File: soc/nordic/common/Kconfig.peripherals
Comment (anangl) [inline soc/nordic/common/Kconfig.peripherals:1]: This "soc: common: nordic_nrf:" in the commit title is outdated and a bit confusing now. Just "soc: nordic:" should be fine.

File: dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi [hunk 1]
```diff
@@ -261,6 +261,30 @@ timer24: timer@ce000 {
 	prescaler = <0>;
 };
 
+pwm20: pwm@d2000 {
+	compatible = "nordic,nrf-pwm";
+	status = "disabled";
+	reg = <0xd2000 0x1000>;
```
Comment (anangl) [inline dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi:267]: Please add also corresponding entries in `validate_base_addresses.c`.

File: soc/nordic/common/Kconfig.peripherals
Comment (nika-nordic) [reply soc/nordic/common/Kconfig.peripherals:1]: done

File: dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi (hunk 1 above)
Comment (nika-nordic) [reply dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi:267]: done

**Review by anangl — DISMISSED** [review]

**Review by anangl — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**